### ISSUE TITLE
WindowServer: Assertion failure on hovering empty dropdown menus

### ISSUE THREAD

**Issue by xTibor** [SerenityOS/serenity#1165]
**Reproduction:**
1. Open PaintBrush
2. Click on the Edit menu
3. Move the mouse downwards

**Stacktrace:**

```
USERSPACE(12) ASSERTION FAILED: window
WSWindowManager.cpp:746 in void WSWindowManager::process_mouse_event(WSMouseEvent&, WSWindow*&)

0xdeadc0de  Scheduler::switch_now() +63
0xdeadc0de  Thread::die_if_needed() +120
0xdeadc0de  syscall_handler +926
0xdeadc0de  syscall_asm_entry +28
0x0809d548  raise +22
0x08048320  abort +13
0x0809d4d0  kill +0
0x0804fc34  WSWindowManager::process_mouse_event(WSMouseEvent&, WSWindow*&) +406
0x0805070c  WSWindowManager::event(Core::Event&) +48
0x08076fd6  Core::Object::dispatch_event(Core::Event&, Core::Object*) +86
0x08079108  Core::EventLoop::pump(Core::EventLoop::WaitMode) +502
0x080792c7  Core::EventLoop::exec() +71
0x08048815  main +1097
0x08048bf3  _start +94
```

**Comment by shannonbooth** [comment]
So the problem is actually pretty simple. When we open a menu we do not create a window for an menu with no items, and in a few places the code expects there to be a window.

I am wondering what sort of behavior we are wanting if there is a menu with no items in it... A simple solution to this would be for the MenuManager (and WindowManager while we still have not migrated all menu code to the MenuManager) to just ignore handling events for empty menu's... but I'm not sure that is what we want.

It looks a little strange having a menu selected with no items in it - I'm wondering whether we should just use a placeholder window to somehow better show that the menu is empty (and not just bugged out)? Or maybe in the menubar not show any menus that do not have any containing items?

I don't know what is best at all to be honest.. any thoughts @awesomekling?

**Comment by xTibor** [comment]
Duplicate issue: #1716 
Fixed by: #1717
